google/cloud/internal/retry_policy.h

```diff
@@ -0,0 +1,151 @@
+
+  explicit LimitedErrorCountRetryPolicy(int maximum_failures)
+      : failure_count_(0), maximum_failures_(maximum_failures) {}
+
+  LimitedErrorCountRetryPolicy(LimitedErrorCountRetryPolicy&& rhs) noexcept
+      : LimitedErrorCountRetryPolicy(rhs.maximum_failures_) {}
+  LimitedErrorCountRetryPolicy(LimitedErrorCountRetryPolicy const& rhs) noexcept
+      : LimitedErrorCountRetryPolicy(rhs.maximum_failures_) {}
+
+  std::unique_ptr<BaseType> clone() const override {
+    return std::unique_ptr<BaseType>(
+        new LimitedErrorCountRetryPolicy(maximum_failures_));
+  }
+  bool Exhausted() const override { return failure_count_ > maximum_failures_; }
+
+ protected:
+  void OnFailureImpl() override { ++failure_count_; }
+
+ private:
+  int failure_count_;
+  int maximum_failures_;
+};
+
+/**
+ * Implement a simple "keep trying for this time" retry policy.
+ */
+template <typename StatusTypeParam, typename RetryablePolicyParam>
+class LimitedTimeRetryPolicy
+    : public RetryPolicy<StatusTypeParam, RetryablePolicyParam> {
+ public:
+  using BaseType = RetryPolicy<StatusTypeParam, RetryablePolicyParam>;
+  using typename BaseType::RetryablePolicy;
+  using typename BaseType::StatusType;
+
+  /**
+   * Constructor given a `std::chrono::duration<>` object.
+   *
+   * @tparam DurationRep a placeholder to match the `Rep` tparam for @p
+   *     duration's type. The semantics of this template parameter are
+   *     documented in `std::chrono::duration<>` (in brief, the underlying
+   *     arithmetic type used to store the number of ticks), for our purposes it
+   *     is simply a formal parameter.
+   * @tparam DurationPeriod a placeholder to match the `Period` tparam for @p
+   *     duration's type. The semantics of this template parameter are
+   *     documented in `std::chrono::duration<>` (in brief, the length of the
+   *     tick in seconds, expressed as a `std::ratio<>`), for our purposes it is
+   *     simply a formal parameter.
+   * @param maximum_duration the maximum time allowed before the policy expires,
+   *     while the application can express this time in any units they desire,
+   *     the class truncates to milliseconds.
+   */
+  template <typename DurationRep, typename DurationPeriod>
+  explicit LimitedTimeRetryPolicy(
+      std::chrono::duration<DurationRep, DurationPeriod> maximum_duration)
+      : maximum_duration_(std::chrono::duration_cast<std::chrono::milliseconds>(
+            maximum_duration)),
+        deadline_(std::chrono::system_clock::now() + maximum_duration_) {}
+
+  LimitedTimeRetryPolicy(LimitedTimeRetryPolicy&& rhs) noexcept
+      : LimitedTimeRetryPolicy(rhs.maximum_duration_) {}
+  LimitedTimeRetryPolicy(LimitedTimeRetryPolicy const& rhs)
+      : LimitedTimeRetryPolicy(rhs.maximum_duration_) {}
+
+  std::unique_ptr<BaseType> clone() const override {
+    return std::unique_ptr<BaseType>(
+        new LimitedTimeRetryPolicy(maximum_duration_));
+  }
+  bool Exhausted() const override {
```

Comment:
Should this be `IsExhausted()`? Or I guess I've missed the boat on this as it's an `override`...

Reply:
Well, the base class is part of the PR, so we could (and I have) change it.

Note that the style guide is silent on this, and there is no consistency, consider `grpc::Status::ok()`, or `std::istream::good()`, `std::istream::eof()` (though there is also `std::istream::is_open()` but that is because `std::istream::open()` is also a thing).  We seem to have more `VerbProperty()`-type predicates in the code, so I am going with it.